lte/gateway/c/core/oai/test/amf/test_amf_encode_decode.cpp

```diff
@@ -842,6 +879,115 @@ TEST(test_dnn, test_amf_validate_dnn) {
   int rc = amf_validate_dnn(&amf_ctx, dnn_string, &idx, ue_sent_dnn);
   EXPECT_TRUE(rc == RETURNok);
 }
+
+TEST(test_pdu_negative, test_unknown_pdu_session_type) {
+  amf_nas_message_t msg = {};
+
+  // build uplinknastransport //
+  // uplink nas transport(pdu session request)
+  uint8_t pdu[44] = {0x7e, 0x00, 0x67, 0x01, 0x00, 0x15, 0x2e, 0x01, 0x01,
+                     0xc1, 0xff, 0xff, 0x95, 0xa1, 0x28, 0x01, 0x00, 0x7b,
+                     0x00, 0x07, 0x80, 0x00, 0x0a, 0x00, 0x00, 0x0d, 0x00,
+                     0x12, 0x01, 0x81, 0x22, 0x01, 0x01, 0x25, 0x09, 0x08,
+                     0x69, 0x6e, 0x74, 0x65, 0x72, 0x6e, 0x65, 0x74};
+
+  uint32_t len = sizeof(pdu) / sizeof(uint8_t);
+
+  ULNASTransportMsg pdu_sess_est_req;
+  bool decode_res = false;
+  memset(&pdu_sess_est_req, 0, sizeof(ULNASTransportMsg));
+
+  decode_res = decode_ul_nas_transport_msg(&pdu_sess_est_req, pdu, len);
```

Comment:
Please add a check that decode was successful

Reply:
Addressed review comment.